src/main/java/nextstep/subway/domain/util/LinePath.java

```diff
@@ -0,0 +1,86 @@
+package nextstep.subway.domain.util;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.Sections;
+import nextstep.subway.domain.Station;
+import nextstep.subway.domain.exception.ExceptionCode;
+import nextstep.subway.domain.exception.LinePathException;
+
+public class LinePath {
+	private Sections sections = new Sections();
+	private WeightedMultigraph<Station, DefaultWeightedEdge> graph = new WeightedMultigraph(DefaultWeightedEdge.class);;
+
+	public LinePath(List<Line> lines) {
+		sections.addAll(getSections(lines));
+		setGraph(lines);
+	}
+
+	public List<Station> searchPath(Station source, Station target) {
+		validFilter(source, target);
+		validExist(source, target);
+
+		DijkstraShortestPath dijkstraShortestPath = new DijkstraShortestPath(graph);
+		Optional<GraphPath> path = Optional.ofNullable(dijkstraShortestPath.getPath(source, target));
```

Comment:
```suggestion
		GraphPath path = Optional.ofNullable(dijkstraShortestPath.getPath(source, target)).orElseThrow(....);
```

이런식으로 구현하는 게 더 좋다고 생각합니다 😅